### ISSUE TITLE
Task lintKotliner does not respect ignored directories on 5.2.0

### ISSUE THREAD

**Issue by xRomZak** [jeremymailen/kotlinter-gradle#474]
A workaround which doesn't work anymore in 5.2.0

```
subprojects {
  tasks.withType<LintTask> {
    this.source = this.source.minus(fileTree("build/generated")).asFileTree
  }
  tasks.withType<FormatTask> {
    this.source = this.source.minus(fileTree("build/generated")).asFileTree
  }
}
```

**Comment by rmarquis** [comment]
I worked around this by disabling KtLint rules for generated code directly in the `.editorconfig` file:

```
[build/generated/**/*]
ktlint = disabled
```

See [KtLint FAQ](https://pinterest.github.io/ktlint/latest/faq/#how-do-i-disable-ktlint-for-generated-code)

**Comment by xRomZak** [comment]
thx, it helped. remove a previous workaround completely

**Comment by xRomZak** [comment]
@rmarquis added a suggested snippet to the `.editorconfig`, but can't get it working together with a set of "disabled" rules.
All the disabled rules are detected again ...

``
ktlint_standard_multiline-if-else = disabled
ktlint_standard_no-wildcard-imports = enabled
ktlint_standard_argument-list-wrapping = disabled
ktlint_standard_class-signature = disabled
ktlint_standard_function-signature = disabled
ktlint_standard_discouraged-comment-location = disabled
ktlint_standard_value-argument-comment = disabled
ktlint_standard_value-parameter-comment = disabled
ktlint_standard_parameter-argument-comment = disabled
ktlint_standard_multiline-expression-wrapping = disabled
ktlint_standard_no-empty-first-line-in-class-body = disabled
ktlint_standard_string-template-indent = disabled
ktlint_standard_if-else-wrapping = disabled
ktlint_standard_indent = disabled
ktlint_standard_parameter-wrapping = disabled
ktlint_standard_parameter-list-wrapping = disabled
ktlint_standard_property-naming = disabled
ktlint_standard_chain-method-continuation = disabled
```

**Comment by jasonab** [comment]
I had to add the source sections under my global rules:

```
root = true

[*.{kt,kts}]
charset = utf-8
end_of_line = lf
indent_size = 4
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true
ktlint_code_style = ktlint_official

# default IntelliJ IDEA style, same as alphabetical, but with "java", "javax", "kotlin" and alias imports in the end of the imports list
ij_kotlin_imports_layout = *, java.**, javax.**, kotlin.**, ^

# it's just too many commas
ij_kotlin_allow_trailing_comma_on_call_site = false
ktlint_chain_method_rule_force_multiline_when_chain_operator_count_greater_or_equal_than = 5
ktlint_class_signature_rule_force_multiline_when_parameter_count_greater_or_equal_than = 3

[src/generated/**/*.{kt,kts}]
ktlint = disabled

[build/generated/**/*.{kt,kts}]
ktlint = disabled

```

**Comment by MyDogTom** [comment]
I'm pretty sure it's because of the recent PR that switches to using `afterEvaluate`. https://github.com/jeremymailen/kotlinter-gradle/pull/472 Some time ago, I was advised against using `afterEvaluate` because it has some drawbacks. Sorry, I can't remember the exact arguments.

Anyway, as another workaround, I believe that moving the `tasks.withType<LintTask>` and `tasks.withType<FormatTask>` configurations into the `project.afterEvaluate { ... }` block should help. 

In my project, I use a convention plugin to configure `kotlinter-gradle`. The following change helped.

```diff
private fun configureKtLint(project: Project) {
    project.plugins.apply("org.jmailen.kotlinter")
    project.extensions.getByType(KotlinterExtension::class.java).apply {
        reporters = arrayOf("checkstyle", "html")
    }

-    project.tasks.withType(LintTask::class.java).configureEach { lintTask ->
-        lintTask.source = lintTask.source.minus(project.fileTree("build/generated")).asFileTree
-    }
-    project.tasks.withType(FormatTask::class.java).configureEach { lintTask ->
-        lintTask.source = lintTask.source.minus(project.fileTree("build/generated")).asFileTree
-    }
+    project.afterEvaluate { afterEvaluateProject ->
+        afterEvaluateProject.tasks.withType(LintTask::class.java).configureEach { lintTask ->
+            lintTask.source = lintTask.source.minus(project.fileTree("build/generated")).asFileTree
+       }
+       afterEvaluateProject.tasks.withType(FormatTask::class.java).configureEach { lintTask ->
+            lintTask.source = lintTask.source.minus(project.fileTree("build/generated")).asFileTree
+        }
+    }
}
```

**Comment by MyDogTom** [comment]
Btw, `.editorconfig` workaround hasone disadvantage. Any changes to `build/generated` directory will invalidate LintTask results because `build/generated` folder is still included in task inputs.

**Comment by knuterikl** [comment]
I resolved this issue by using exclude like this:

```kts
tasks.withType<LintTask> {
    exclude { it.file.path.contains("/build/generatedSources/") }
}

tasks.withType<FormatTask> {
    exclude { it.file.path.contains("/build/generatedSources/") }
}
```

**Comment by jeremymailen** [comment]
Thanks for the working config. Updating the Readme.
But I agree with everyone on both points
1. Excluding generated source ought to be a first class feature available in the `kotlinter { }` config.
2. We should update the setup of lint and format tasks to follow best practices and Gradle 9 friendly.

Maybe @Karn and I can work on it, but also open to contributions.
